gtsam/hybrid/tests/testHybridFactorGraph.cpp

```diff
@@ -376,67 +361,271 @@ TEST(HybridFactorGraph, ToDecisionTreeFactor) {
+  }
 
-  CHECK(result.second);
-  //  GTSAM_PRINT(*result.second);  // HybridFactorGraph
-  EXPECT_LONGS_EQUAL(3, result.second->size());
+};
+
+/* ****************************************************************************/
+// Test if we can incrementally do the inference
+TEST_UNSAFE(DCGaussianElimination, Incremental_inference) {
+  Switching switching(3);
+
+  IncrementalHybrid incrementalHybrid;
+
+  HybridFactorGraph graph1;
+
+  graph1.push_back(switching.linearizedFactorGraph.dcGraph().at(0));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(0));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(1));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(2));
+
+  // Create ordering.
+  Ordering ordering;
+  ordering += X(1);
+  ordering += X(2);
+
+  incrementalHybrid.update(graph1, ordering);
+
+  auto hybridBayesNet = incrementalHybrid.hybridBayesNet_;
+  CHECK(hybridBayesNet);
+  EXPECT_LONGS_EQUAL(2, hybridBayesNet->size());
+  EXPECT(hybridBayesNet->at(0)->frontals() == KeyVector{X(1)});
+  EXPECT(hybridBayesNet->at(0)->parents() == KeyVector({X(2), M(1)}));
+  EXPECT(hybridBayesNet->at(1)->frontals() == KeyVector{X(2)});
+  EXPECT(hybridBayesNet->at(1)->parents() == KeyVector({M(1)}));
+
+  auto remainingFactorGraph = incrementalHybrid.remainingFactorGraph_;
+  CHECK(remainingFactorGraph);
+  EXPECT_LONGS_EQUAL(1, remainingFactorGraph->size());
+
+  auto discreteFactor_m1 = *dynamic_pointer_cast<DecisionTreeFactor>(remainingFactorGraph->discreteGraph().at(0));
+  EXPECT(
+      discreteFactor_m1.keys() == KeyVector({M(1)}));
+
+  HybridFactorGraph graph2;
+
+  graph2.push_back(switching.linearizedFactorGraph.dcGraph().at(1)); // p(x3 | x2, m2)
+  graph2.push_back(switching.linearizedFactorGraph.gaussianGraph().at(3));
+
+  // Create ordering.
+  Ordering ordering2;
+  ordering2 += X(2);
+  ordering2 += X(3);
+
+  incrementalHybrid.update(graph2, ordering2);
+
+  auto hybridBayesNet2 = incrementalHybrid.hybridBayesNet_;
+  CHECK(hybridBayesNet2);
+  EXPECT_LONGS_EQUAL(2, hybridBayesNet2->size());
+  EXPECT(hybridBayesNet2->at(0)->frontals() == KeyVector{X(2)});
+  EXPECT(hybridBayesNet2->at(0)->parents() == KeyVector({X(3), M(2), M(1)}));
+  EXPECT(hybridBayesNet2->at(1)->frontals() == KeyVector{X(3)});
+  EXPECT(hybridBayesNet2->at(1)->parents() == KeyVector({M(2), M(1)}));
+
+  auto remainingFactorGraph2 = incrementalHybrid.remainingFactorGraph_;
+  CHECK(remainingFactorGraph2);
+  EXPECT_LONGS_EQUAL(1, remainingFactorGraph2->size());
+
+  auto discreteFactor = dynamic_pointer_cast<DecisionTreeFactor>(remainingFactorGraph2->discreteGraph().at(0));
+  EXPECT(
+      discreteFactor->keys()
+          == KeyVector({M(2), M(1)}));
+
+  ordering.clear();
+  ordering += X(1);
+  ordering += X(2);
+  ordering += X(3);
+
+  // Now we calculate the actual factors using full elimination
+  HybridBayesNet::shared_ptr actualHybridBayesNet;
```

Comment:
I would not call these `actual` here but `expected`

Reply:
done